### PR TITLE
Fix get version to return the same version during and after replay

### PR DESCRIPTION
**Pull request by meiliang86**

### CODE REVIEW

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java [hunk 1]
```diff
@@ -4097,6 +4100,43 @@ public void testGetVersion() {
...
+
+    @Override
+    public String execute(String taskList) {
+      // Test adding a version check in replay code.
+      if (!getVersionExecuted.contains(taskList + "-test_change_2")) {
+        getVersionExecuted.add(taskList + "-test_change_2");
+        Workflow.sleep(Duration.ofHours(1));
+      } else {
+        int version2 = Workflow.getVersion("test_change_2", Workflow.DEFAULT_VERSION, 1);
+        System.out.println("version = " + version2);
```
Comment (emrahs) [inline src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: nit: Consider asserting the value of `version2`

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java [hunk 2]
```diff
@@ -4097,6 +4100,43 @@ public void testGetVersion() {
...
+        Workflow.sleep(Duration.ofHours(1));
+
+        int version3 = Workflow.getVersion("test_change_2", Workflow.DEFAULT_VERSION, 1);
+        System.out.println("version = " + version3);
```
Comment (emrahs) [inline src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: nit: same here

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java [hunk 3]
```diff
@@ -4037,6 +4037,7 @@ public String execute(String taskList) {
 
       // Test adding a version check in non-replay code.
       int version = Workflow.getVersion("test_change", Workflow.DEFAULT_VERSION, 1);
+      assertEquals(version, 1);
```
Comment (vancexu) [inline src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: maybe we can remove the rest if-else?

Comment (meiliang86) [reply src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: Yes removed.

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java (hunk 2 above)
Comment (meiliang86) [reply src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: Removed this print. It's for my own debugging purpose.

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java (hunk 1 above)
Comment (meiliang86) [reply src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: Removed this print. It's for my own debugging purpose.

File: src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java [hunk 4]
```diff
@@ -311,7 +311,8 @@ int getVersion(String changeId, DataConverter converter, int minSupported, int m
   }
 
   private void validateVersion(String changeID, int version, int minSupported, int maxSupported) {
-    if (version < minSupported || version > maxSupported) {
+    if ((version < minSupported || version > maxSupported)
+        && version != WorkflowInternal.DEFAULT_VERSION) {
```
[the remark this answers is not in the document]
Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java]: Sorry forogot to add more context. Please refer to https://github.com/uber/cadence-java-client/issues/466 and the unit test I added.

**Comment by meiliang86** [comment]
Fixes https://github.com/uber/cadence-java-client/issues/466

File: src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java]: The change in MarkerHandler.java in the real fix. The change here is a side effect.

**Pull request merged**